doc: modem: documents LTE fallback mode (#6243)
nrfconnect/sdk-nrf: doc/nrf/libraries/modem/lte_lc.rst

Additional configurations related to these features can be found in the API documentation for the link controller.

Connection fallback mode
========================
It is possible to try to switch between LTE-M and NB-IoT after a certain time period if a connection has not been established.

> This comes useful in case connection to either network becomes unavailable.

Reviewer: ```suggestion
This is useful when the connection to either of these networks becomes unavailable.
```

> Switching period between the network modes is configurable.

Reviewer: ```suggestion
You can also configure the switching period between the network modes.
```

> If a connection cannot be established by using the fallback mode the library will report an error.

Reviewer: ```suggestion
If a connection cannot be established by using the fallback mode, the library reports an error.
You can use the following configuration options to configure the connection fallback mode:
```
